src/util/compare.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+namespace nest {
+namespace mc {
+namespace util {
+
+template<typename E>
+struct lessthan {
```

Comment:
and this is the templated functor, which is templated on another templated functor.